modules/rostests/apitests/acpi/Bus_PDO_EvalMethod.c

```diff
@@ -0,0 +1,495 @@
+/*
+ * PROJECT:     ReactOS API Tests
+ * LICENSE:     LGPL-2.1-or-later (https://spdx.org/licenses/LGPL-2.1-or-later)
+ * PURPOSE:     Unit Tests for acpi!Bus_PDO_EvalMethod
+ * COPYRIGHT:   Copyright 2023 Dmitry Borisov (di.sean@protonmail.com)
+ */
+
+/* INCLUDES *******************************************************************/
+
+#include <apitest.h>
+
+#define WIN32_NO_STATUS
+#include <ndk/rtlfuncs.h>
+
+#define UNIT_TEST
+#include <acpi.h>
+
+#include <acpiioct.h>
+
+/* KERNEL DEFINITIONS (MOCK) **************************************************/
+
+#define DPRINT(...)  do { if (0) DbgPrint(__VA_ARGS__); } while (0)
+#define DPRINT1(...) do { if (0) DbgPrint(__VA_ARGS__); } while (0)
+
+typedef struct _IO_STACK_LOCATION
+{
+    union
+    {
+        struct
+        {
+            ULONG OutputBufferLength;
+            ULONG InputBufferLength;
+        } DeviceIoControl;
+    } Parameters;
+} IO_STACK_LOCATION, *PIO_STACK_LOCATION;
+
+typedef struct _IRP
+{
+    union
+    {
+        PVOID SystemBuffer;
+    } AssociatedIrp;
+    IO_STATUS_BLOCK IoStatus;
+
+    IO_STACK_LOCATION MyStack;
+} IRP, *PIRP;
+
+#define NonPagedPool 1
+static
+PVOID
+ExAllocatePoolWithTag(ULONG PoolType, SIZE_T NumberOfBytes, ULONG Tag)
+{
+    PVOID *Mem;
```

Comment:
What about using `ULONG_PTR*` (or PULONG_PTR)  type here? This would avoid the casts for the Mem[i] below, while keeping the whole thing correct. (Exception would be a cast for the return)

Reply:
Fixed